### ISSUE TITLE
Unconverged fits do not return the best sampled fit

### ISSUE THREAD

**Issue by mdimura** [lmfit/lmfit-py#679]
If the sampling is aborted because `max_nfev` was reached, the returned MinimizerResult.params contains the latest samled parameters instead of the best available fit.
Here is the [gist](https://gist.github.com/mdimura/871858b83399d66a73e236ddb4cf1e2c) with the sample code.
```python
import lmfit
import numpy as np 

def residual(pars, x, data=None):
    """Model a decaying sine wave and subtract data."""
    vals = pars.valuesdict()
    amp = vals['amp']
    per = vals['period']
    shift = vals['shift']
    decay = vals['decay']

    if abs(shift) > np.pi/2:
        shift = shift - np.sign(shift)*np.pi
    model = amp * np.sin(shift + x/per) * np.exp(-x*x*decay*decay)
    if data is None:
        return model
    return model - data

def chi2r(pars, x, data):
    return np.square(residual(pars, x, data)).sum()/x.size

np.random.seed(1)

x = np.linspace(0.0, 250., 10000)
noise = np.random.normal(scale=0.7215, size=x.size)

params = lmfit.Parameters()
params.add('amp', value=13.0)
params.add('period', value=2)
params.add('shift', value=0.0)
params.add('decay', value=0.02)

data = residual(params, x) + noise

#scramble
for name in params:
    params[name].value *= np.random.uniform(0.1,10.0)

nSteps = 10000
bestChi2r = chi2r(params, x, data)
def progress(p, i, resi, *_):
    global bestChi2r 
    bestChi2r = min(bestChi2r, resi)
out = lmfit.minimize(chi2r, params, args=(x, data), method='basinhopping', max_nfev=nSteps, iter_cb=progress)
lastChi2 = chi2r(out.params, x, data)
print(f'Result chi2r = {lastChi2}')
print(f'True best chi2r = {bestChi2r}')
#print(lmfit.fit_report(out))
```
My result:
```
Result chi2r = 9.360
True best chi2r = 0.519
```
I would expect them to match.

**Comment by newville** [comment]
@mdimura  Hm, this could be a duplicate of #665.   Then again, if the fit aborts early because it reaches the maximum number of evaluations that you set, I don't see why you have any valid expectations of what the result should be.  But far more importantly, go back and read the instructions about when and how to raise an issue.  Do not ignore those instructions again.

**Comment by mdimura** [comment]
According to the [docs](https://lmfit.github.io/lmfit-py/fitting.html#lmfit.minimizer.MinimizerResult.params), should contain, I quote:

> The best-fit parameters resulting from the fit.

Which made me think, this is actually a bug. It would also be much more practical to report the best fit since for many real-life applications these global minimizers will not fully converge in any reasonable time but still find the best or very good solution relatively fast.
Sorry if I violated any contribution guidelines, I tried to follow them.

**Comment by newville** [comment]
@mdimura 

> Sorry if I violated any contribution guidelines, I tried to follow them.

It's entirely possible that this problem was fixed already in #665, as I mentioned earlier.  You did not give version information as instructed (with a warning of "DO NOT IGNORE") when creating an issue, so I don't know whether this applies or not.  It is demotivating to continue saying this, but: follow the instructions.   You didn't violate contributor guidelines, this is a basic "did not give a complete enough report to act on".

**Comment by mdimura** [comment]
My bad, the lmfit version is 1.0.1, which is from May 7, and the fix is from Aug 24, so yes, it's probably the old version, sorry I bothered.
